stabilize.md

```diff
@@ -49,39 +49,137 @@ Flap Semantics
+    rule <k> Flap . exception ~>  Flap . catch => Flap . pop  ... </k>
+    rule <k> Flap . exception ~> (Flap . FS    => .)          ... </k>
+      requires FS =/=K catch
+```
 
-    syntax FlapStep ::= "kick" Int Int
- // ----------------------------------
+- kick(uint lot, uint bid) returns (uint id)
+- Starts a new surplus auction for a lot amount
 
+**TODO** The `end` field of the Bid (TIME +Int 172800) is supposed to be two days from the current time.
+
+```k
+    syntax FlapAuthStep ::= "kick" Int Int
+ // --------------------------------------
+    rule <k> Flap . kick LOT BID => Vat . move MSGSENDER THIS LOT ~> KICK +Int 1  ... </k>
+         <msg-sender> MSGSENDER </msg-sender>
+         <this> THIS </this>
+         <currentTime> NOW </currentTime>
+         <flap-bids> M => M[KICK +Int 1 <- Bid(BID, LOT, MSGSENDER, 0, NOW +Int 172800)] </flap-bids>
+         <flap-kicks> KICK => KICK +Int 1 </flap-kicks>
+         <flap-live> 1 </flap-live>
+```
+
+- tend(uint id, uint lot, uint bid)
+- Places a bid made by the user. Refunds the previous bidder's bid.
+
+**TODO** Flap.tend needs to call Gem.move. We don't have Gem yet.
+`<k> Flap . tend ID LOT BID => Gem . move MSGSENDER GUY CURBID ~> Gem . move MSGSENDER THIS (BID -Int CURBID) ... </k>`
+**TODO** (TIC +Int 10800) Represents TIC plus 3 hours.
+
+```k
     syntax FlapStep ::= "tend" Int Int Int
  // --------------------------------------
+    rule <k> Flap . tend ID LOT BID => . ... </k>
+         <msg-sender> MSGSENDER </msg-sender>
+         <this> THIS </this>
+         <currentTime> NOW </currentTime>
+         <flap-bids> ... ID |-> Bid( (CURBID => BID), CURLOT, (GUY => MSGSENDER), (TIC => TIC +Int 10800), END ) ... </flap-bids>
+         <flap-live> 1 </flap-live>
+      requires GUY =/=Int 0
+       andBool (TIC >Int NOW orBool TIC ==Int 0)
+       andBool END >Int NOW
+       andBool LOT ==Int CURLOT
+       andBool BID  >Int CURBID
+       andBool BID *Int ilk_init >=Int CURBID *Int 1050000000000000000000000000
+```
 
-    syntax FlapStep ::= "deal" Int
- // ------------------------------
+- deal(uint id)
+- Settles an auction, rewarding the lot to the highest bidder and burning their bid
 
-    syntax FlapStep ::= "cage" Int
- // ------------------------------
+**TODO** Flap.deal calls Gem.burn
+`<k> Flap . deal ID => Vat . move THIS GUY LOT ~> Gem . burn THIS BID ... </k>`
 
-    syntax FlapStep ::= "yank" Int
- // ------------------------------
+```k
+    syntax FlapStep ::= "deal" Int [klabel(FlapDeal),symbol]
+ // --------------------------------------------------------
+    rule <k> Flap . deal ID => Vat . move THIS GUY LOT ... </k>
+         <this> THIS </this>
+         <currentTime> NOW </currentTime>
+         <flap-bids> ... (ID |-> Bid( BID, LOT, GUY, TIC, END ) => .Map) ... </flap-bids>
+         <flap-live> 1 </flap-live>
+      requires (TIC <Int NOW andBool TIC =/=Int 0)
+        orBool  END <Int NOW
```

Comment:
I think this should read `(TIC =/=Int 0) andBool (TIC <Int NOW orBool END <Int NOW)`.